src/core/events/metadata.rs

```diff
@@ -0,0 +1,54 @@
+#![allow(dead_code)] // FIXME
+
+use anyhow::Result;
+
+use super::*;
+use crate::module::ModuleId;
+
+/// Metadata attached to a series of event.
+///
+/// We reuse here the Event object internally as we're storing the exact same
+/// kind of data (map of ModuleId to custom structs) with the same set of
+/// capabilities (JSON conversion, marshalling, etc). However we redefine our
```

Comment:
Doesn't this approach limit the scope of data that we can store here?
Most of the uses we've found to the metadata section are "common", where as it's unlikely other modules will need to store metadata associated with them (maybe you've thought of a use-case I fail to see?).

Therefore, this is a limitation originated by internal implementation details (e.g: trying to reuse code) that ends up causing most of the information to unnecessarily be inside a "common".